### PR TITLE
Fix slice viewer bug caused by switching to `Log` or `SymmetricLog` on masked data

### LINKED ISSUE #33883: Unhandled exception when changing colorscale to log with empty bins present and autoscaling on in Sliceviewer
(1) Load in `CNCS_7860_event.nxs` - zoom into a region that contains zero bins
(2) Change colorscale to Log

```
Traceback (most recent call last):
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\cbook\__init__.py", line 287, in process
    func(*args, **kwargs)
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\colorbar.py", line 551, in update_normal
    self.draw_all()
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\colorbar.py", line 578, in draw_all
    self._process_values()
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\colorbar.py", line 1144, in _process_values
    b = self.norm.inverse(b)
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\colors.py", line 1554, in inverse
    raise ValueError("Invalid vmin or vmax")
ValueError: Invalid vmin or vmax
```

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR fixes a bug on the sliceviewer caused by zooming in to data which is all masked and then changing the normalisation to `Log` or `SymmetricLog10`.
It also fixes a similar bug (reproducible only on a conda build) causing an error when switching to `Log` after zooming in to a region with barely any data.

**To test:**
From a conda environment:

1. Load in `CNCS_7860_event.nxs` from `<build_dir>/ExternalData/Testing/Data/SystemTest/` folder
2. Zoom out, and then zoom into a region with no data.
3. The `Log` and `SymmetricLog10` colorscale options should be disabled and have a useful tooltip
4. Click the Home button, and these options should now be enabled and with no tooltip
5. Zoom into a region that contains almost zero bins (where only one small line can be seen on the plot)
6. Change colorscale to log, and there should be no error message. There might be a warning but this is ok

Fixes #33883

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@robertapplin The target branch here is currently `main` - should it be `release-next`?

**Comment by robertapplin** [comment]
> @robertapplin The target branch here is currently `main` - should it be `release-next`?

Yes, thanks for spotting that

**Comment by sf1919** [comment]
I think this PR also fixes #33411

**Comment by sf1919** [comment]
@robertapplin and myself have also tested it against #33310 and thats been fixed too.

File: docs/source/release/v6.4.0/Workbench/SliceViewer/Bugfixes/33883.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+- Using the Log or SymmetricLog10 colorbar normalisation options on masked data is now disabled to prevent an error.
```
Comment (sf1919) [inline docs/source/release/v6.4.0/Workbench/SliceViewer/Bugfixes/33883.rst:1]: In light of this also fixing #33310 please can the release note be updated?

Comment (robertapplin) [reply docs/source/release/v6.4.0/Workbench/SliceViewer/Bugfixes/33883.rst:1]: I believe the error in #33310 was caused by our switch to matplotlib 3.5, and so is not present in v6.3. I will check and make sure

Comment (robertapplin) [reply docs/source/release/v6.4.0/Workbench/SliceViewer/Bugfixes/33883.rst:1]: I'm mistaken, it is in v6.3. I'll add a release note for it

**Review by sf1919 — APPROVED** [review]
This solution tested well and has in fact resolved two other issues. I have reviewed the code and it looks good.

**Pull request merged**